app/crud.py

```diff
@@ -0,0 +1,304 @@
+    if db_user:
+        db_user.is_active = True
+        db.commit()
+        db.refresh(db_user)
+    return db_user
+
+
+# -------------------------------------------------------------------------------------------------
+# User Type specific CRUD operations, # of functions = 5
+# -------------------------------------------------------------------------------------------------
+
+def get_user_type_by_id(db: Session, user_type_id: int) -> models.UserType:
+    """
+    This function queries the database for the UserType with the given user_type_id
+    :param db: The database session
+    :param user_type_id: The user_type_id
+    :return: The UserType with the given user_type_id
+    """
+    return db.query(models.UserType).filter(models.UserType.user_type_id == user_type_id).first()
+
+
+def get_user_type_by_name(db: Session, name: str) -> models.UserType:
+    """
+    This function queries the database for the UserType with the given name
+    :param db: The database session
+    :param name: The user_type_name
+    :return: The UserType with the given name
+    """
+    return db.query(models.UserType).filter(models.UserType.name == name).first()
+
+
+def get_user_types(db: Session, skip: int = 0, limit: int = 10) -> list[models.UserType]:
+    """
+    This function queries the database for the UserType with the given skip and limit boundaries
+    and returns a list of UserTypes
+    :param db: The database session
+    :param skip: Starting index of the list, 0 by default
+    :param limit: Ending index (skip + limit), 10 by default
+    :return: List[Type[models.UserType]]:
+    """
+    return db.query(models.UserType).offset(skip).limit(limit).all()
+
+
+def create_user_type(db: Session, user_type: schemas.UserTypeCreate) -> models.UserType:
+    """
+    This function creates a new UserType
+    :param db: The database session
+    :param user_type: schemas.UserTypeCreate
+    :return: The created UserType
+    """
+    db_user_type = models.UserType(
+        **user_type.dict()
+    )
+    db.add(db_user_type)
+    db.commit()
+    db.refresh(db_user_type)
+    return db_user_type
+
+
+def delete_user_type(db: Session, user_type_id: int) -> models.UserType:
+    """
+    This function deletes a User Type based on the given user_type_id
+    :param db: The database session
+    :param user_type_id: The user_type_id
+    :return: Deleted UserType
+    """
+
+    db_user_type = db.query(models.UserType).filter(models.UserType.user_type_id == user_type_id).first()
+    db.delete(db_user_type)
+    db.commit()
+    return db_user_type
+
+
+# -------------------------------------------------------------------------------------------------
+# Course specific CRUD operations, # of functions = 5
+# -------------------------------------------------------------------------------------------------
+
+def get_course_by_id(db: Session, course_id: int) -> models.Course:
```

Comment:
Split functions in separate files according to the model they are supposed to modify. Imagine if we would have 30, 50 or 100 tables. Tracking all the changes in a single file would be so difficult.